### ISSUE TITLE
Scrap Releases in favor of just using Games

### ISSUE THREAD

**Issue by connorshea** [connorshea/ContinueFromCheckpoint#152]
The idea behind Releases was that each Game would have one or more Releases. For example:

Game: Portal 2
Releases:
- Portal 2 for Windows
- Portal 2 for macOS
- Portal 2 for Linux
- Portal 2 for PlayStation 3
- Portal 2 for Xbox 360

## Problems with the current Releases system
- I intend to use Wikidata as my data source here, and releases don't map cleanly to Wikidata items. Portal 2 has one Wikidata item ([Q279446](https://www.wikidata.org/wiki/Q279446)), with the platforms Windows, macOS, Linux, PS3, and Xbox 360. It doesn't specify the developers or release dates for a given platform.
  - It'd be nice to be able to more easily use data from Wikidata (and also PCGamingWiki) to fill out the database.
- If you own Portal 2 on Steam, you have it on Windows, macOS, and Linux. Should you add 3 different releases to your library in this case? That seems pretty dumb and inconvenient.
- Most games have the same data across platforms, e.g. release dates and developers/publishers. There are, of course, exceptions (Portal 2, for example, didn't release on Linux originally in 2011).
- You can own the same game on Windows in two or more different stores (for example, Stardew Valley on GOG.com and Steam). In order to track something like that, the game needs to be represented differently from how it is now.
- When representing games on the games#index, a lot of game data is pulled from the game's releases, which requires loading a lot of records in many cases. This is pretty likely to lead to performance problems.
- It's confusing to users what Games are vs. what Releases are, and it's kind of hard to explain. The sharing of data between games and releases would probably end up being a point of confusion as well.

## Problems with using only Games

- How should this be handled in the database/Rails models?
  - For example, do we want to track per-platform release dates/developers/publishers still?
  - If so, would JSONB be able to handle this? What would the performance impact be?
- A lot of code will need to be refactored.
- How do we represent the games in a user's library, for example how does the user set the platform they own the game on? Or even the store (e.g. Steam, Origin, GOG.com)?
- No more special handling of compilation games, though that was a poorly-conceived-of and never-properly-implemented feature anyway.

## TL;DR

I think I'll remove Releases and switch to just having Games, it'll require a large amount of refactoring, and have problems of its own, but ultimately it should simplify things and make the site easier to use. 

Rest in Peace Releases, December 2018 - February 2019. 🕯
